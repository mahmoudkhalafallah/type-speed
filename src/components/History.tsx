import React from 'react'
// eslint-disable-next-line no-unused-vars
import { RouteComponentProps } from '@reach/router'

declare interface Props extends RouteComponentProps {
    historyId?: string
}

const History: React.FC<Props> = ({ historyId }) => {
    console.log(historyId)

    return <>
    </>
}

export default History