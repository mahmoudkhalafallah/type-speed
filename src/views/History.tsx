import React, { useEffect, useState } from 'react'
import { HISTORY_URL } from '../constants'
import { RouteComponentProps } from "@reach/router" // eslint-disable-line no-unused-vars
import styled from 'styled-components'

declare interface Props extends RouteComponentProps {
    historyId?: string
}

const RecordList = styled.ul`
list-style: none;
padding: 0;
`

const RecordItem = styled.li`
display: flex;
justify-content: space-between;
font-size: 20px;
`
const RecordNum = styled.b`
margin-left: 15px;
`

const History: React.FC<Props> = ({ historyId }) => {

    const [gameRecordsHistoryData, setGameRecordsHistoryData] = useState({ "wpm": "", "cp": "" })
    const [dataLoaded, setDataLoaded] = useState(false)


    useEffect(() => {
        if (historyId && historyId.length > 0) {
            fetch(`${HISTORY_URL}/${historyId}`)
                .then(res => res.json())
                .then((data: { "wpm": string, "cp": string }) => {
                    setGameRecordsHistoryData(data)
                })
                .catch(err => err)
                .finally(() => {
                    setDataLoaded(true)
                })
        }
    }, [historyId])

    return <>
        {(gameRecordsHistoryData.wpm && gameRecordsHistoryData.cp) ?
            <RecordList data-testid='recordList'>
                <RecordItem>Words per minute: <RecordNum>{gameRecordsHistoryData.wpm}</RecordNum>WPM</RecordItem>
                <RecordItem>Completion Percentage: <RecordNum>{gameRecordsHistoryData.cp}</RecordNum> %</RecordItem>
            </RecordList> :
            dataLoaded ? <h2>No Records</h2> : <span>Loading</span>
        }
    </>
}

export default History