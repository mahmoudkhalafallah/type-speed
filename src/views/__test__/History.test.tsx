import React from 'react'
import History from '../History'
import renderer from 'react-test-renderer'
import 'jest-fetch-mock'

test('History rendered correctly', () => {
    const component = renderer.create(
        <History historyId='m0nww' />,
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})
