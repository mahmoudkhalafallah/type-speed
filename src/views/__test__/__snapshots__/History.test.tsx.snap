// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History rendered correctly 1`] = `
<span>
  Loading
</span>
`;
